doc/src/manual/constructors.md

```diff
@@ -119,23 +119,27 @@ existence by a call to one of the inner constructor methods provided with the ty
 some degree of enforcement of a type's invariants.
 
 If any inner constructor method is defined, no default constructor method is provided: it is presumed
-that you have supplied yourself with all the inner constructors you need. The default constructor
-is equivalent to writing your own inner constructor method that takes all of the object's fields
-as parameters (constrained to be of the correct type, if the corresponding field has a type),
-and passes them to `new`, returning the resulting object:
+that you have supplied yourself with all the inner constructors you need.
+
+The default constructor is equivalent to writing one or two of your own constructor methods that take
+all of the object's fields as parameters.  One is an inner constructor method without type constraints
+on its parameters that passes the parameters to `new`, returning the resulting object.  The other
+default constructor method is provided if there are any type constraints on the fields. This method
+is an inner constructor method if the type is not generic and similarly passes the parameters to `new`.
```

Comment:
I'm not sure I understand what you mean by "generic," are you talking about the case `struct S1{T}` and the distinction between `S1(x)` and `S1{Int}(x)`?

We may also want to take the opportunity to disambiguate parameters in the sense of parametric types from arguments supplied to the constructor. In Julia I would tend to use arguments and not parameters to describe methods, and reserve the term parameter for parametric types.

Reply:
+1 that it's better to use the term arguments for the variables that are passed into a function. I know that in PL usually we use parameters and arguments to refer to the callsite and the definition site (in that order? I can never remember), but I think Tim is right that in julia docs we tend to prefer `arguments` for both, and reserver parameters for discussions about type parameters on a DataType.

Also +1 that i'm not quite clear on the "generic" part of this.

@timholy: Can you provide a suggestion for what we should say here instead?

Reply:
It's actually a bit nontrivial. "We do the minimum thing required for everything to work in various circumstances" is the closest to the truth, but that's a bit unsatisfying. This PR seems devoted mostly to the non-parametric case, but the parametric case is a bit different.